From 06da440bf502548e4aadb899bcbadee0a2319f9b Mon Sep 17 00:00:00 2001
From: Eric Long <i@hack3r.moe>
Date: Sun, 28 Jan 2024 17:41:06 +0800
Subject: [PATCH] Fix Rust target triple for riscv64

Rust target triple for riscv64 is riscv64gc-unknown-linux-gnu rather
than Clang's riscv64-unknown-linux-gnu.

Change-Id: I2f19101bd654f2f71392a65ce619c30570cdea49
---
 build/config/rust.gni | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/build/config/rust.gni b/build/config/rust.gni
index c0b8450c9b786..c3e01344b078d 100644
--- a/build/config/rust.gni
+++ b/build/config/rust.gni
@@ -177,6 +177,8 @@ if (is_linux || is_chromeos) {
     cpu = "i686"
   } else if (cpu == "x64") {
     cpu = "x86_64"
+  } else if (cpu == "riscv64") {
+    cpu = "riscv64gc"
   }
   rust_abi_target = cpu + "-unknown-linux-gnu"
 } else if (is_android) {
-- 
2.39.2

